src/Controller/JobController.php

```diff
@@ -0,0 +1,259 @@
+     * @return Response
+     */
+    public function readJobsByName(string $name): Response
+    {
+        try {
+            $this->jobValidator->nameIsValid($name);
+            $rows = $this->jobsRepository->selectByName($name);
+            return new JsonResponse([
+                'results' => [
+                    'error' =>false,
+                    $rows
+                ]
+            ]);
+        } catch (\Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+  
+    /**
+     * read a job by the substring of the name
+     *
+     * @param  mixed $name
+     * @return Response
+     */
+    public function readJobsByNameLike(string $name): Response
+    {
+        try {
+            $this->jobValidator->nameIsValid($name);
+            $rows = $this->jobsRepository->selectByNameLike($name);
+            return new JsonResponse([
+                'results' => [
+                    'error' =>false,
+                    $rows
+                ]
+            ]);
+        } catch (\Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * updates a job from the database
+     *
+     * @param  mixed $id
+     * @param  mixed $request
+     * @return Response
+     */
+    public function updateJob(int $id, Request $request): Response
+    {
+        try {
+            $this->jobValidator->idIsValid($id);
+            $name = $request->get('name');
+            $this->jobValidator->nameIsValid($name);
+            $requestParams = $request->query->all();
+            $updateResult = $this->jobsRepository->update($id, $requestParams);
+
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => false,
+                        'rows_updated' => $updateResult
+                    ]
+                ]
+            );
+        } catch (\Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * delete a job from the database
+     *
+     * @param  mixed $id
+     * @return Response
+     */
+    public function deleteJob(int $id): Response
+    {
+        try {
+            $this->jobValidator->idIsValid($id);
+            $deletedId = null;
+            $jobToDelete = $this->jobsRepository->find($id);
+            if (!empty($compToDelete)) {
```

Comment:
$compToDelete is not declared anywhere. The delete will never work with this code. Update the code to be able to delete a job.